CMakeLists.txt

```diff
@@ -267,6 +267,8 @@ if(USE_OPENCV)
       set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wl,--no-undefined")
     endif()
   endif()
+  add_executable(im2rec "tools/im2rec.cc")
+  target_link_libraries(im2rec mxnet_static ${OpenCV_LIBS} ${mxnet_LINKER_LIBS} dmlc)
```

Comment:
check cpp unit tests CMakeLists.txt, mxnet_static should be wrapped with BEGIN_WHOLE_ARCHIVE and END_WHOLE_ARCHIVE

Reply:
Thanks, fixed. Anything else that catches your eye?

Reply:
Nope. Ping mew when it's done and I will merge.

Reply:
Oh wait, that should be expansion macros like in unit tests:

  if(UNITTEST_STATIC_LINK)
    target_link_libraries(${PROJECT_NAME}_unit_tests
      ${GTEST_LIBRARY}
      ${BEGIN_WHOLE_ARCHIVE} mxnet_static ${END_WHOLE_ARCHIVE}
      dmlc
      ${mxnet_LINKER_LIBS}
      ${pslite_LINKER_LIBS}
      )
  else()

Reply:
Should definitely have: ${BEGIN_WHOLE_ARCHIVE} mxnet_static ${END_WHOLE_ARCHIVE}.  I've run into that issue a few times now ;-).  If you don't put this the linker seems to optimize out all your operators, so you'll get some FCompute, or FInferShape not found error at runtime.

Aren't we safe to always statically link the lib here though?  Any idea why this is turned of for MSVS for the unit tests? (I'm assuming you guys ran into issues at some point).

Reply:
I left it off for MSVC only because I don't have a Windows machine and it was getting errors during Windows CI build back then. If it can be made to work, then that's fine.

Reply:
(If we are referring to the unit tests)

Reply:
We are.  Thanks for the background Chris..